crates/cosmos/cosmos-chain-components/src/impls/channel/channel_handshake_message.rs

```diff
@@ -84,6 +86,7 @@ where
     Chain: HasMessageType
         + HasChannelIdType<Counterparty, ChannelId: Display>
         + HasPortIdType<Counterparty, PortId: Display>
+        + HasConnectionIdType<Counterparty>
         + CanRaiseAsyncError<ClientError>,
```

Comment:
I think we also need to update the `connection_hops` in the body to use the connection ID in the payload.

Reply:
do you need to discuss this comment? I need some context to understand.

Reply:
It is inside the method body for `build_channel_open_try_message`, which is unchanged so I can't comment directly on it. Basically, we need to update the message builder to use the correct connection ID.

Reply:
Ah yes ! Actually, we can remove the message builders for cosmos at ibc-starknet repo and use the ones from Hermes-SDK.

Reply:
Fixed in e115f40693cd0418a0cba54fae44481546b7781d.